### PR TITLE
JAMES-3057 make MailboxId of Mailbox object immutable

### PR DESCRIPTION
**Pull request by Arsnael**
For this, we need a pre-step of changing the existing `MailboxMapper.create(Mailbox)` to `MailboxMapper.create(MailboxPath, uidValidity)`, returning a Mailbox object instead of a MailboxId.

From there, the rest was not complicated...

### CODE REVIEW

**Comment by Arsnael** [comment]
```
[e83651eba0fd4df096dd46ac4c6c820771260e03] [ERROR] Failed to execute goal on project apache-james-backends-jpa: Could not resolve dependencies for project org.apache.james:apache-james-backends-jpa:jar:3.5.0-SNAPSHOT: Failed to collect dependencies at com.google.guava:guava:jar:25.1-jre -> org.codehaus.mojo:animal-sniffer-annotations:jar:1.14: Failed to read artifact descriptor for org.codehaus.mojo:animal-sniffer-annotations:jar:1.14: Could not transfer artifact org.codehaus:codehaus-parent:pom:4 from/to central (https://repo.maven.apache.org/maven2): /root/.m2/repository/org/codehaus/codehaus-parent/4/codehaus-parent-4.pom.part (No such file or directory) -> [Help 1]
```
......... test this please

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/MailboxMapper.java [hunk 1]
```diff
@@ -44,6 +44,11 @@
      */
     MailboxId create(Mailbox mailbox) throws MailboxException;
 
+    /**
+     * Create a {@link Mailbox} with the given {@link MailboxPath} and uid to the underlying storage
+     */
+    Mailbox create(MailboxPath mailboxPath, long uidValidity) throws MailboxException;
```
Comment (chibenwa) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/MailboxMapper.java]: I would be picky, I would ask for UidValidity strong typing ;-)

Comment (Arsnael) [reply mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/MailboxMapper.java]: Was not the primary goal but while we are at it, sure :)

Comment (Arsnael) [reply mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/MailboxMapper.java]: In an other PR though

Comment (Arsnael) [reply mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/MailboxMapper.java]: #3132

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**